src/Application/PresenterFactory.php

```diff
@@ -44,21 +44,23 @@ public function __construct(Nette\DI\Container $container)
 	public function createPresenter($name)
 	{
 		$class = $this->getPresenterClass($name);
-		if (count($services = $this->container->findByType($class)) === 1) {
-			$presenter = $this->container->createService($services[0]);
-			$tags = $this->container->findByTag(Nette\DI\Extensions\InjectExtension::TAG_INJECT);
-			if (empty($tags[$services[0]])) {
-				$this->container->callInjects($presenter);
-			}
-		} else {
+		$services = $this->container->findByType($class);
+		if (count($services) > 1) {
+			throw new InvalidPresenterException("Multiple services of type $class found: " . implode(', ', array_keys($services)) . '.');
+
+		} elseif (!$services) {
+			$rc = new \ReflectionClass($this->container);
+			@unlink($rc->getFileName() . '.meta'); // @ file may not exists
```

Comment:
Yes, there should be. But you certainly must not do this in production. If someone is using presenters without the `Presenter` suffix, this will get triggered on every request. Or am I wrong?